Add browse images for the RTC-S1-STATIC product

gshiroma opened the pull request:
This PR adds browse images for the RTC-S1-STATIC product

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -373,6 +375,103 @@ def save_browse(imagery_list, browse_image_filename,
+    gdal_band = gdal_ds.GetRasterBand(1)
+    band_image = np.asarray(gdal_band.ReadAsArray(), dtype=np.float32)
+    is_valid = np.isfinite(band_image)
+    if alpha_channel is None:
+        alpha_channel = np.asarray(is_valid,
+                                   dtype=np.float32)
```
```suggestion
    alpha_channel = np.asarray(is_valid,
                               dtype=np.float32)
```
The variable `alpha_channel` in this function is always `None` (see around line 409), so I do not see the necessity of the if statement here.
Same for the function `save_browse_imagery()`

gshiroma replied:
I fixed that for static layers. For imagery, that if statement is inside a for loop and we only need to assign values to alpha channel once.

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -373,6 +375,103 @@ def save_browse(imagery_list, browse_image_filename,
+    logger.info(f'        max ({BROWSE_IMAGE_MAX_PERCENTILE}% percentile):'
+                f' {vmax}')
+
+    band_image = ((band_image - vmin) / (vmax - vmin))
```
```suggestion
    band_image = (band_image - vmin) / (vmax - vmin)
```
No need to wrap the parenthesis here.

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -373,6 +375,103 @@ def save_browse(imagery_list, browse_image_filename,
+                       band_image,
+                       band_image,
+                       alpha_channel))
+  
```
```suggestion

```
get rid of the whitespace

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -1726,11 +1825,27 @@ def run_single_job(cfg: RunConfig):
+                    product_type == STATIC_LAYERS_PRODUCT_TYPE):
+                browse_image_filename = \
+                    os.path.join(output_dir_bursts, f'{burst_product_id}.png')
+                save_browse_static(radar_grid_file_dict_filenames[0],
```
It seems the static image is generated based on `radar_grid_file_dict_filenames[0]`, and the list is actually the values of the dictionary `radar_grid_file_dict`. This can cause inconsistencies on the source layer for the static browse image because:

- The first element in the list `radar_grid_file_dict_filenames` can change depending on the `save_*` flags (e.g. `save_local_inc_angle`, `save_incidence_angle`, etc.).
- It would be probably okay for our SASes (i.e. using python >= 3.7) but the older python3 does not preserve order of the key-value pair in dict.

I think it would be probably okay for the static layer production for OPERA, unless we make changes on the `save_*` flags in runconfig. But I foresee that the source layers for the static layer can be different depending on the runconfig setting.

Maybe it would be a good idea to clarify what static layer is used in `browse_image_filename`?

gshiroma replied:
Thank you, @seongsujeong , I'm handling this here:

https://github.com/opera-adt/RTC/pull/56/commits/7504e4410fc035475308cc40d2a3bd25015f2dae

seongsujeong commented on src/rtc/rtc_s1_single_job.py:
```diff
@@ -2010,15 +2142,15 @@ def get_radar_grid(geogrid, dem_interp_method_enum, product_id,
+    incidence_angle_raster = _create_raster_obj(
+        output_dir, product_id, LAYER_NAME_INCIDENCE_ANGLE,
+        gdal.GDT_Float32, shape, radar_grid_file_dict,
+        output_obj_list, save_incidence_angle, extension)
```
I think it would be more readable if the order of `_create_raster_obj()` is consistent with the order of the parameters `save_*` flags in this function. I see your intention as to place the local incidence angle as the first key-value pair in the dictionary `radar_grid_file_dict`. That being case, I think you can swap the locations of `save_incidence_angle` and `save_local_inc_angle` in the function header.

seongsujeong reviewed (commented):
Here are my initial comments - Seongsu

gshiroma commented on src/rtc/rtc_s1.py:
```diff
@@ -17,7 +17,8 @@
                                    snap_coord,
                                    get_radar_grid,
-                                   save_browse,
+                                   save_browse_imagery,
```
RTC is radiometric terrain correction. One may think that RTC is the map of the RTC flattening factors (which is the output of the RTC module). I may be wrong, but I think the term "imagery" is better understood as satellite/product imagery.

seongsujeong replied:
Okay. That being case let's leave it as it is.

gshiroma commented:
Thank you, @seongsujeong , for this quick review on a Saturday! I just addressed your comments. Please, let me know if I missed anything or if you have any other comments or concerns.

seongsujeong reviewed (approved):
LGTM. I see some issue with the naming of the function `get_radar_grid()` but it is not significant enough to delay the merging of this PR. I will submit a PR to fix that after the point delivery.

gshiroma commented:
Thank you, @seongsujeong
